Backend/Users/WebApiService/RegistrationService.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Backend.Users.Repository;
+using Model.Users;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Backend.Users.WebApiService
+{
+    public class RegistrationService
+    {
+        IPatientRepository patientRepository;
+
+        public RegistrationService(IPatientRepository patientRepository)
+        {
+            this.patientRepository = patientRepository;
+        }
+
+        public Patient Register(Patient patient)
+        {
+            if (!PatientExists(patient.Id, patient.Username))
+            {
+                return patientRepository.Create(patient);
+            }
+
+            return null;
+        }
+
+        public bool PatientExists(string id, string username)
+        {
+            List<Patient> patients = patientRepository.GetAll().ToList();
+            return patients.Any(patient => patient.Username != null && patient.Username.Equals(username) || patient.Id.Equals(id));
+        }
+        public bool ExistsById(string id)
+        {
+            return patientRepository.ExistsById(id);
+        }
+
+        public Patient ExistsByUsername(string username)
+        {
+            List<Patient> patients = patientRepository.GetAll().ToList();
+            bool exists = patients.Any(patient => patient.Username != null && patient.Username.Equals(username));
+            if (exists)
+            {
+                return patients.FirstOrDefault(patient => patient.Username.Equals(username));
+            }
+            return null;
```

Comment:
I would recommend changing the return statement if not found to raising a not found exception.